fe/src/main/java/org/apache/doris/alter/MaterializedViewHandler.java

```diff
@@ -457,8 +462,23 @@ private RollupJobV2 createMaterializedViewJob(String mvName, String baseIndexNam
                                                + "duplicate table");
             }
             Column newMVColumn = new Column(baseColumn);
+
             newMVColumn.setIsKey(mvColumnItem.isKey());
             newMVColumn.setAggregationType(mvAggregationType, mvColumnItem.isAggregationTypeImplicit());
+            newMVColumn.setDefineExpr(mvColumnItem.getDefineExpr());
```

Comment:
Firstly, check the define expr is forbidden or not. Secondly, set define expr in column